plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineWebProblemsTest.java

```diff
@@ -25,48 +25,48 @@
 import org.junit.Assert;
 import org.junit.Test;
 
-public class AppEngineWebBlacklistTest {
+public class AppEngineWebProblemsTest {
   
   @Test
   public void testBlacklisted() {
-    Assert.assertTrue(AppEngineWebBlacklist.contains("application"));
+    Assert.assertTrue(AppEngineWebProblems.contains("application"));
   }
  
   @Test
   public void testContains() {
-    assertFalse(AppEngineWebBlacklist.contains("foo"));
+    assertFalse(AppEngineWebProblems.contains("foo"));
   }
   
   @Test
   public void testContains_nullArg() {
-    assertFalse(AppEngineWebBlacklist.contains(null));
+    assertFalse(AppEngineWebProblems.contains(null));
   }
   
   @Test(expected = NullPointerException.class)
   public void testGetBlacklistElementMessage_nullArg() {
-    AppEngineWebBlacklist.getBlacklistElementMessage(null);
+    AppEngineWebProblems.getDeprecatedElementMessage(null);
   }
   
   @Test(expected = IllegalArgumentException.class)
   public void testGetBlacklistElementMessage_elementNotInBlacklist() {
-    AppEngineWebBlacklist.getBlacklistElementMessage("test");
+    AppEngineWebProblems.getDeprecatedElementMessage("test");
   }
   
   @Test
   public void testGetBlacklistElementMessage() {
     assertEquals("Project ID should be specified at deploy time",
-      AppEngineWebBlacklist.getBlacklistElementMessage("application"));
+      AppEngineWebProblems.getDeprecatedElementMessage("application"));
```

Comment:
I think it's not deprecated actually. We recently added the feature in the new Maven and Gradle plugins to allow reading the `<application>` and `<version>` values from `appengine-web.xml` (in addition to allow getting the values from the `gcloud` config or `pom.xml`). In that sense, it's just that our IDE plugin will never use these values (but I think we may update our IDE plugin to allow users to use the values in `appengine-web.xml`.)

Reply:
FYI https://github.com/GoogleCloudPlatform/app-maven-plugin/pull/306/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR18